File: generator/src/test/resources/kotlin/expected/sunday/RequestMethodsTest/test-request-method-generation.output.kt
package io.test.service

import io.outfoxx.sunday.MediaType
import io.outfoxx.sunday.Operation
import io.outfoxx.sunday.OperationSpec
import io.outfoxx.sunday.Transport
import io.outfoxx.sunday.http.Method
import io.outfoxx.sunday.http.Request
import io.outfoxx.sunday.http.Response
import io.outfoxx.sunday.operation
import io.test.PatchableTest
import io.test.Test
import kotlin.Unit
import kotlin.collections.List

public class API<Req : Request>(
  public val transport: Transport<Req>,
  public val defaultContentTypes: List<MediaType> = listOf(MediaType.JSON),
  public val defaultAcceptTypes: List<MediaType> = listOf(MediaType.JSON),
) {
  public fun fetchTest(): Operation<Unit, Test, Req> = this.transport.operation<Unit, Test, Req>(
    OperationSpec(
      method = Method.Get,
      pathTemplate = "/tests",
      acceptTypes = this.defaultAcceptTypes
    )
  )

  public fun putTest(body: Test): Operation<Test, Test, Req> =
      this.transport.operation<Test, Test, Req>(
    OperationSpec(
      method = Method.Put,
      pathTemplate = "/tests",
      body = body,
      contentTypes = this.defaultContentTypes,
      acceptTypes = this.defaultAcceptTypes
    )
  )

  public fun postTest(body: Test): Operation<Test, Test, Req> =
      this.transport.operation<Test, Test, Req>(
    OperationSpec(
      method = Method.Post,
      pathTemplate = "/tests",
      body = body,
      contentTypes = this.defaultContentTypes,
      acceptTypes = this.defaultAcceptTypes
    )
  )

  public fun patchTest(body: Test): Operation<Test, Test, Req> =
      this.transport.operation<Test, Test, Req>(
    OperationSpec(
      method = Method.Patch,
      pathTemplate = "/tests",
      body = body,
      contentTypes = this.defaultContentTypes,
      acceptTypes = this.defaultAcceptTypes
    )
  )

  public fun deleteTest(): Operation<Unit, Unit, Req> = this.transport.operation<Unit, Unit, Req>(
    OperationSpec(
      method = Method.Delete,
      pathTemplate = "/tests"
    )
  )

  public fun headTest(): Operation<Unit, Unit, Req> = this.transport.operation<Unit, Unit, Req>(
    OperationSpec(
      method = Method.Head,
      pathTemplate = "/tests"
    )
  )

  public fun optionsTest(): Operation<Unit, Unit, Req> = this.transport.operation<Unit, Unit, Req>(
    OperationSpec(
      method = Method.Options,
      pathTemplate = "/tests"
    )
  )

  public fun patchableTest(body: PatchableTest): Operation<PatchableTest, Test, Req> =
      this.transport.operation<PatchableTest, Test, Req>(
    OperationSpec(
      method = Method.Patch,
      pathTemplate = "/tests2",
      body = body,
      contentTypes = this.defaultContentTypes,
      acceptTypes = this.defaultAcceptTypes
    )
  )

  public suspend fun requestTest(): Req = this.transport
    .transportRequest(
      method = Method.Get,
      pathTemplate = "/request",
      acceptTypes = this.defaultAcceptTypes
    )

  public suspend fun responseTest(): Response = this.transport
    .transportResponse(
      method = Method.Get,
      pathTemplate = "/response",
      acceptTypes = this.defaultAcceptTypes
    )
}

File: generator/src/test/resources/swift/expected/RequestMethodsTest/test-request-method-generation.output.swift
import Sunday

public final class API<TransportType : Transport> : Sendable {

  public static var problemTypes: [ProblemRegistration] {
    return []
  }
  public let transport: TransportType
  public let defaultContentTypes: [MediaType]
  public let defaultAcceptTypes: [MediaType]

  public init(
    transport: TransportType,
    defaultContentTypes: [MediaType] = [.json],
    defaultAcceptTypes: [MediaType] = [.json],
    problemTypes: [ProblemRegistration] = API.problemTypes
  ) {
    self.transport = transport
    self.defaultContentTypes = defaultContentTypes
    self.defaultAcceptTypes = defaultAcceptTypes
    problemTypes.forEach { $0.register(on: transport) }
  }

  public func fetchTest() throws -> Sunday.Operation<Empty, Test, TransportType> {
    return Sunday.Operation(
      transport: self.transport,
      spec: Sunday.OperationSpec(
        method: .get,
        pathTemplate: "/tests",
        pathParameters: nil,
        queryParameters: nil,
        body: Empty.none,
        contentTypes: nil,
        acceptTypes: self.defaultAcceptTypes,
        headers: nil
      )
    )
  }

  public func putTest(body: Test) throws -> Sunday.Operation<Test, Test, TransportType> {
    return Sunday.Operation(
      transport: self.transport,
      spec: Sunday.OperationSpec(
        method: .put,
        pathTemplate: "/tests",
        pathParameters: nil,
        queryParameters: nil,
        body: body,
        contentTypes: self.defaultContentTypes,
        acceptTypes: self.defaultAcceptTypes,
        headers: nil
      )
    )
  }

  public func postTest(body: Test) throws -> Sunday.Operation<Test, Test, TransportType> {
    return Sunday.Operation(
      transport: self.transport,
      spec: Sunday.OperationSpec(
        method: .post,
        pathTemplate: "/tests",
        pathParameters: nil,
        queryParameters: nil,
        body: body,
        contentTypes: self.defaultContentTypes,
        acceptTypes: self.defaultAcceptTypes,
        headers: nil
      )
    )
  }

  public func patchTest(body: Test) throws -> Sunday.Operation<Test, Test, TransportType> {
    return Sunday.Operation(
      transport: self.transport,
      spec: Sunday.OperationSpec(
        method: .patch,
        pathTemplate: "/tests",
        pathParameters: nil,
        queryParameters: nil,
        body: body,
        contentTypes: self.defaultContentTypes,
        acceptTypes: self.defaultAcceptTypes,
        headers: nil
      )
    )
  }

  public func deleteTest() throws -> Sunday.Operation<Empty, Void, TransportType> {
    return Sunday.Operation(
      transport: self.transport,
      spec: Sunday.OperationSpec(
        method: .delete,
        pathTemplate: "/tests",
        pathParameters: nil,
        queryParameters: nil,
        body: Empty.none,
        contentTypes: nil,
        acceptTypes: self.defaultAcceptTypes,
        headers: nil
      )
    )
  }

  public func headTest() throws -> Sunday.Operation<Empty, Void, TransportType> {
    return Sunday.Operation(
      transport: self.transport,
      spec: Sunday.OperationSpec(
        method: .head,
        pathTemplate: "/tests",
        pathParameters: nil,
        queryParameters: nil,
        body: Empty.none,
        contentTypes: nil,
        acceptTypes: self.defaultAcceptTypes,
        headers: nil
      )
    )
  }

  public func optionsTest() throws -> Sunday.Operation<Empty, Void, TransportType> {
    return Sunday.Operation(
      transport: self.transport,
      spec: Sunday.OperationSpec(
        method: .options,
        pathTemplate: "/tests",
        pathParameters: nil,
        queryParameters: nil,
        body: Empty.none,
        contentTypes: nil,
        acceptTypes: self.defaultAcceptTypes,
        headers: nil
      )
    )
  }

  public func patchableTest(body: PatchableTest) throws -> Sunday.Operation<PatchableTest, Test, TransportType> {
    return Sunday.Operation(
      transport: self.transport,
      spec: Sunday.OperationSpec(
        method: .patch,
        pathTemplate: "/tests2",
        pathParameters: nil,
        queryParameters: nil,
        body: body,
        contentTypes: self.defaultContentTypes,
        acceptTypes: self.defaultAcceptTypes,
        headers: nil
      )
    )
  }

  public func requestTest() async throws -> TransportType.Request {
    return try await self.transport.transportRequest(
      method: .get,
      pathTemplate: "/request",
      pathParameters: nil,
      queryParameters: nil,
      body: Empty.none,
      contentTypes: nil,
      acceptTypes: self.defaultAcceptTypes,
      headers: nil
    )
  }

  public func responseTest() async throws -> TransportType.Response {
    return try await self.transport.transportResponse(
      method: .get,
      pathTemplate: "/response",
      pathParameters: nil,
      queryParameters: nil,
      body: Empty.none,
      contentTypes: nil,
      acceptTypes: self.defaultAcceptTypes,
      headers: nil
    )
  }

}
